src/event.rs

```diff
@@ -106,6 +106,49 @@ pub enum Event {
     #[cfg(feature = "xv")]
     /// The event is issued from the `XVideo` extension.
     Xv(xv::Event),
+
+    /// The event was not recognized, it was likely issued from a disabled extension.
+    Unresolved(UnresolvedEvent),
+}
+
+/// an event was not recognized as part of the core protocol or any enabled extension
+pub struct UnresolvedEvent {
+    raw: *mut xcb_generic_event_t,
+}
+
+impl BaseEvent for UnresolvedEvent {
+    const EXTENSION: Option<Extension> = None;
+    const NUMBER: u32 = u32::MAX;
```

Comment:
This is the one thing I'm not sure about, I couldn't find any usages of the constant on `BaseEvent` within `xcb`, I hope a dummy value is acceptable here.

Reply:
`BaseEvent::NUMBER` is in fact not used. For the time being, I don't see anything better than this.